lib/pbench/test/unit/server/test_requests.py

```diff
@@ -151,43 +153,52 @@ def test_mismatched_md5sum_header(client, caplog, server_config):
 
             filename = "log.tar.xz"
             datafile = Path("./lib/pbench/test/unit/server/fixtures/upload/", filename)
-            controller = socket.gethostname()
 
-            with open(datafile, "rb") as data_fp:
+            with datafile.open("rb") as data_fp:
                 response = client.put(
-                    f"{server_config.rest_uri}/upload/ctrl/{controller}",
+                    TestUpload.gen_uri(server_config, filename),
                     data=data_fp,
                     headers={
                         "Authorization": "Bearer " + auth_token,
-                        "filename": filename,
+                        # Content-Length header set automatically
                         "Content-MD5": "md5sum",  # Wrong md5 hash
                     },
                 )
             assert response.status_code == 400
-            assert (
-                response.json.get("message")
-                == f"md5sum check failed for {filename}, upload failed"
+            assert response.json.get("message").startswith(
+                "MD5 checksum does not match Content-MD5 header"
             )
+            for record in caplog.records:
+                assert record.levelname not in ("DEBUG", "ERROR", "CRITICAL")
+            assert caplog.records[-1].levelname == "WARNING"
 
     @staticmethod
     @pytest.mark.parametrize("bad_extension", ("test.tar.bad", "test.tar", "test.tar."))
-    def test_bad_extension_upload(client, bad_extension, caplog, server_config):
+    def test_bad_extension_upload(
+        client, bad_extension, pytestconfig, caplog, server_config
+    ):
         with client:
             auth_token = get_pbench_token(client, server_config)
 
-            expected_message = "File extension not supported. Only .xz"
-            response = client.put(
-                f"{server_config.rest_uri}/upload/ctrl/{socket.gethostname()}",
-                headers={
-                    "Authorization": "Bearer " + auth_token,
-                    "filename": bad_extension,
-                    "Content-MD5": "md5sum",
-                },
-            )
-            assert response.status_code == 400
+            expected_message = "File extension not supported, must be .tar.xz"
+            tmp_d = pytestconfig.cache.get("TMP", None)
+            datafile = Path(tmp_d, bad_extension)
+            datafile.write_text("compressed tar ball")
+            with datafile.open("rb") as data_fp:
+                response = client.put(
+                    TestUpload.gen_uri(server_config, bad_extension),
+                    data=data_fp,
+                    headers={
+                        "Authorization": "Bearer " + auth_token,
+                        # Content-Length header set automatically
+                        "Content-MD5": "md5sum",  # Ignored
+                    },
+                )
+            assert response.status_code == 415
```

Comment:
415? That doesn't sound right. "Media type" has a fairly specific meaning, and file name suffix doesn't fit very well.

Reply:
The only plausible alternative that I could come up with was [422](https://tools.ietf.org/html/rfc4918#section-11.2); otherwise, I think we're relegated to 400, which is probably OK if the message is clear.

Reply:
Moved to #2287.